features/step_definitions/finance/providers/billing_settings_step.rb

```diff
@@ -71,3 +71,11 @@
 Given /^master is( not)? billing tenants$/ do |master_billing_disabled|
   ThreeScale.stubs(master_billing_enabled?: !master_billing_disabled)
 end
+
+Given "a provider has charging configured and visible" do
+  step %(a provider exists)
+  step %(the provider has a default paid application plan)
+  step %(the provider is charging)
+  step %(the provider has "finance" switch visible)
+  step %(a buyer signed up to the provider)
```

Comment:
Yikes 😄 If we could avoid reusing steps as they were functions, it would be perfect. Usually what I do is just copy paste what's in those steps and then try to extract helper methods. This is OK as a hack but I would rather have duplicated code than reusing steps.

Reply:
Why?
I actually find duplicated steps or steps that do similar things rather confusing.

I imagine that that happens because it's easier to just write a new step that search through the code whether there is a step already that does what I want.

To me it makes sense to have small "building blocks" and compose them as the use case requires.